### PR TITLE
add 'getPluginFromType()' method in PluginConnector

### PR DESCRIPTION
**Pull request by Heikman**
add 'getPluginFromType()' method in PluginConnector.

See the discussion in #1376.

### CODE REVIEW

File: src/libPMacc/include/pluginSystem/PluginConnector.hpp [hunk 1]
```diff
@@ -180,6 +181,19 @@ namespace PMacc
             }
         }
 
+        template<typename Plugin>
+        Plugin* getPluginFromType()
```
Comment (psychocoderHPC) [inline src/libPMacc/include/pluginSystem/PluginConnector.hpp]: Please add a description to this function.
What is if there are more than one plugin with the same type inside the list named `plugins`?

Comment (psychocoderHPC) [inline src/libPMacc/include/pluginSystem/PluginConnector.hpp]: I checked if it is possible that more than one plugin can have the same type. It is possible e.g., in PIConGPU the plugin `Visualisation<ParticleType,OuputMethod>` is registered multiple times with the same type signature.
This means that the name must be `getFirstPluginFromType()` or it should be return a vector with all instances. The second version is IMO the cleanest solution.

**Comment by Heikman** [comment]
tested with #1376

File: src/libPMacc/include/pluginSystem/PluginConnector.hpp [hunk 2]
```diff
@@ -180,6 +182,27 @@ namespace PMacc
             }
         }
 
+        /**
+         * Get a vector of pointers of all plugin instances from a given type.
```
Comment (ax3l) [inline src/libPMacc/include/pluginSystem/PluginConnector.hpp]: of all *registered* plugin instances

**Comment by Heikman** [comment]
ready for merging?

**Pull request merged**